crypto/dh/dh_ameth.c

```diff
@@ -536,6 +536,60 @@ static int dhx_pkey_import_from(const OSSL_PARAM params[], void *vpctx)
     return dh_pkey_import_from_type(params, vpctx, EVP_PKEY_DHX);
 }
 
+#define dh_bn_dup_check(bn) \
+{ \
+  const BIGNUM *f = DH_get0_##bn(dh); \
+  \
+  if (f != NULL && (dupkey->bn = BN_dup(f)) == NULL) \
+      goto err; \
+}
+
+static DH *dh_dup(const DH *dh)
+{
+    DH *dupkey = NULL;
+
+    /* Do not try to duplicate foreign DH keys */
```

Comment:
Maybe copy flags, version and pad also?

Reply:
pad is always zero. version as well. I'll copy the flags.